### PR TITLE
A simple cache for the furnace recipes, to avoid linear iteration

### PR DESCRIPTION
**Pull request by repo-alt**
(Should help with https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/6416)

### CODE REVIEW

File: src/main/java/gregtech/api/util/GT_ModHandler.java [hunk 1]
```diff
@@ -122,6 +122,7 @@
     public static List<Integer> sSingleNonBlockDamagableRecipeList_warntOutput = new ArrayList<Integer>(50);
     public static List<Integer> sVanillaRecipeList_warntOutput = new ArrayList<Integer>(50);
     public static final List<IRecipe> sSingleNonBlockDamagableRecipeList_verified = new ArrayList<IRecipe>(1000);
+    private static HashMap<GT_ItemStack, ItemStack> smeltingRecipeCache = new LinkedHashMap<>();
```
Comment (Glease) [inline src/main/java/gregtech/api/util/GT_ModHandler.java:125]: Maybe use guava Cache instead? I feel like an unbounded cache will not be *that* helpful and can cause its own problems.

Comment (repo-alt) [reply src/main/java/gregtech/api/util/GT_ModHandler.java:125]: Any hash map will be better than vanilla array, regardless of the size. Guava has it's own hash map implementation?

Comment (Glease) [reply src/main/java/gregtech/api/util/GT_ModHandler.java:125]: See [this page](https://github.com/google/guava/wiki/CachesExplained). The thing I want most is size constraint.

Comment (repo-alt) [reply src/main/java/gregtech/api/util/GT_ModHandler.java:125]: Wow, that's a good stuff. I wish they allowed overriding comparison operator too

**Review by draknyte1 — CHANGES_REQUESTED** [review]
Rename smeltingRecipeCache to sSmeltingRecipeCache.

**Comment by Dream-Master** [comment]
AlkalusToday at 1:31 PM
Looks fine.

**Comment by draknyte1** [comment]
You merged it before I could actually leave the approved review. 
2fast4me

**Comment by Dream-Master** [comment]
@draknyte1 sorry. So should i revert it ?

**Comment by draknyte1** [comment]
No no no. All is good. 👍

**Pull request merged**